inject/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -1026,6 +1067,82 @@ public void accept(Map<String, ClassElement> stringClassElementMap) {
         this.beanFinalized = true;
     }
 
+    private void checkIfShouldLoad(GeneratorAdapter adapter, List<AnnotationPropertyVisitData> beanPropertyVisitData) {
+        for (int i = 0; i < beanPropertyVisitData.size(); i++) {
+            int currentIndex = i;
+            AnnotationPropertyVisitData visitData = beanPropertyVisitData.get(i);
+            MethodVisitData getterVisitData = visitData.getterVisitData;
+            // load 'this'
+            adapter.loadThis();
+            // 1st argument load BeanResolutionContext
+            adapter.loadArg(0);
+            // 2nd argument load BeanContext
+            adapter.loadArg(1);
+            // 2rd argument the field index
+            adapter.push(currentIndex);
+            // push qualifier
+            pushQualifier(adapter, visitData.annotationBeanType, () -> resolveAnnotationArgument(adapter, currentIndex));
+            // invoke getBeanForField
+            pushInvokeMethodOnSuperClass(adapter, GET_BEAN_FOR_ANNOTATION);
+            // cast the return value to the correct type
+            pushCastToType(adapter, getterVisitData.getBeanType().getType());
+            // invoke property getter
+            adapter.invokeVirtual(JavaModelUtils.getTypeReference(getterVisitData.getBeanType()),
+                org.objectweb.asm.commons.Method.getMethod(getterVisitData.getMethodElement().getDescription(false)));
+
+            // storing getter return value to local variable
+            int propertyValueIndex = adapter.newLocal(JavaModelUtils.getTypeReference(getterVisitData.getMethodElement().getReturnType()));
+            adapter.storeLocal(propertyValueIndex);
+            // loading conversion service instance
+            adapter.getStatic(Type.getType(ConversionService.class), "SHARED", Type.getType(ConversionService.class));
```

Comment:
Extract `Type.getType(ConversionService.class)` into a variable it is used a lot